### PR TITLE
Add instructions for contributing and testing locally

### PR DESCRIPTION
**Pull request by nicoleepp**
Link to the rendered README: https://github.com/invenia/TagBotGitLab/blob/ne/add-contributing-instructions/README.md

Some of the changes are just moving sections around

### CODE REVIEW

File: local_test.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
+# This file is intended for local testing when contributing to this repository
+# Do not committ any changes
```
Comment (TylerLoewen) [inline local_test.py:2]:
```suggestion
# Do not commit any changes
```

File: local_test.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+# You will need to generate a Gitlab Personal Access Token to use this
```
Comment (TylerLoewen) [inline local_test.py:3]:
```suggestion
# You will need to generate a GitLab Personal Access Token to use this
```

File: README.md [hunk 3]
```diff
@@ -11,17 +11,31 @@ Our Julia packages are registered in a private Julia package registry hosted on
 Each MR registers a new package or a new version of a Julia package.
 Even though we host a private Registrator deployment, this can be used on any GitLab Julia package registry repository independently from where the MRs originate.
 
-## License
+## Changelogs (Release Notes)
 
-tagbotgitlab is provided under an MIT License.
+TagBotGitlab creates a changelog for each release based on the issues that have been closed and the merge requests that have been merged. Unlike [TagBot](https://github.com/JuliaRegistries/TagBot), TagBotGitLab currently does not support custom release notes or customizable templates.
```
Comment (TylerLoewen) [inline README.md:16]:
```suggestion
TagBotGitLab creates a changelog for each release based on the issues that have been closed and the merge requests that have been merged. Unlike [TagBot](https://github.com/JuliaRegistries/TagBot), TagBotGitLab currently does not support custom release notes or customizable templates.
```

File: README.md [hunk 4]
```diff
@@ -59,19 +73,16 @@ Steps:
 
 This code is tested on GitLab version `11.11.0-ee`.
 
-### Changelogs (Release Notes)
+## Contributing
 
-TagBotGitlab creates a changelog for each release based on the issues that have been closed and the merge requests that have been merged. Unlike [TagBot](https://github.com/JuliaRegistries/TagBot), TagBotGitLab currently does not support custom release notes or customizable templates.
+This package uses the [python-gitlab](https://python-gitlab.readthedocs.io/en/stable/index.html) package to interact with GitLab and it's useful to refer to their documentation when making changes.
+You can test using their API locally by generating a [Personal Access Token](https://docs.gitlab.com/ee/user/profile/personal_access_tokens.html#creating-a-personal-access-token) and setting it in the `local_test.py` script included.
```
Comment (TylerLoewen) [inline README.md:79]:
```suggestion
You can test using their API locally by generating a [Personal Access Token](https://docs.gitlab.com/ee/user/profile/personal_access_tokens.html#creating-a-personal-access-token) and setting it in the included `local_test.py` script.
```

File: README.md [hunk 5]
```diff
@@ -59,19 +73,16 @@ Steps:
 
 This code is tested on GitLab version `11.11.0-ee`.
 
-### Changelogs (Release Notes)
+## Contributing
 
-TagBotGitlab creates a changelog for each release based on the issues that have been closed and the merge requests that have been merged. Unlike [TagBot](https://github.com/JuliaRegistries/TagBot), TagBotGitLab currently does not support custom release notes or customizable templates.
+This package uses the [python-gitlab](https://python-gitlab.readthedocs.io/en/stable/index.html) package to interact with GitLab and it's useful to refer to their documentation when making changes.
```
Comment (TylerLoewen) [inline README.md:78]:
```suggestion
This package uses the [python-gitlab](https://python-gitlab.readthedocs.io/en/stable/index.html) package to interact with GitLab.
It's useful to refer to their documentation when making changes.
```

File: local_test.py [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+for tag in tags:
+    commit = tag.commit["id"]
+    version = tag.name
+
+    release_notes = changelog.get(version, commit)
+    print(release_notes)
+    print("\n-----------------------------------------------------------------------\n")
+
+    # Note the line below will actually set the release notes in the repository used
+    # Should only be used if that is intended behaviour
```
Comment (TylerLoewen) [inline local_test.py:36]:
```suggestion
    # Should only be used if that is the intended behaviour
```

File: README.md (hunk 4 above)
Comment (TylerLoewen) [reply README.md:79]: Might be worth mentioning which scopes are required for the personal access token

Comment (nicoleepp) [reply README.md:79]: It depends on what API calls you are trying to test

**Review by TylerLoewen — APPROVED** [review]

**Pull request merged**